Update `persist_docs` docs for adapter support (#1666)
fishtown-analytics/docs.getdbt.com: website/docs/reference/resource-configs/persist_docs.md

## Support

The `persist_docs` config is supported on the most widely used dbt adapters:
- Postgres
- Redshift
- Snowflake
- BigQuery
- Apache Spark & Databricks

> However, some database limit where and how descriptions can be added to database objects. Those database adapters may not support `persist_docs`, or may offer only partial support.

Reviewer: ```suggestion
However, some databases limit where and how descriptions can be added to database objects. Those database adapters might not support `persist_docs`, or might offer only partial support.
```

Reviewer: suggestion: use `might` instead of `may`. 

motivation: `May` can also mean to have permission like, "you may have your drink now", so `might` is preferred as it removes ambiguity to help make translation/localization easier.

Some known issues and limitations:

<WHCode>

<div warehouse="Databricks">

- Column-level comments require `file_format: delta` (or another "v2 file format")

> - Column-level comments are not supported for models materialized as <Term id="view">views</Term> ([issue](https://github.com/dbt-labs/dbt-spark/issues/372)

Reviewer: ```suggestion
- Column-level comments aren't supported for models materialized as <Term id="view">views</Term> ([issue](https://github.com/dbt-labs/dbt-spark/issues/372))
```

Reviewer: suggestion: use contractions and missing close paren for github issue

motivation: for a more conversational tone (versus formal)

</div>

<div warehouse="Snowflake">

<VersionBlock firstVersion="1.2">

- No known issues

> </VersionBlock>

Reviewer: curious, is this version block required? 

from a content perspective, we shouldn't list "no known issues" in a "known issues and limitations" list. if this version block isn't required, suggest we remove it.

Author: This is a technical limitation of our current implementation for `VersionBlock` + `WHCode`. There _are_ known issues on Snowflake for earlier versions, but as v1.2, they have been resolved. We cannot nest the "Snowflake" WHCode div within a `VersionBlock`, so it must appear regardless. Would we prefer it be empty for v1.2+, or say "no known issues"?

<VersionBlock firstVersion="1.0" lastVersion="1.1">

- Column names that must be quoted, such as column names containing special characters, will cause runtime errors if column-level `persist_docs` is enabled. This is fixed in v1.2.

</VersionBlock>

<VersionBlock lastVersion="0.21">

> - Column-level comments are not supported for models materialized as <Term id="view">views</Term>

Reviewer: ```suggestion
- Column-level comments aren't supported for models materialized as <Term id="view">views</Term>
```

## Support

> The `persist_docs` config is supported on the most widely used dbt adapters: BigQuery, Redshift, Snowflake, Postgres, and Apache Spark/Databricks. Some databases impose limitations on the types of descriptions that can be added to database objects, so their adapters may not support `persist_docs`, or may offer only partial support. For example, column-level comments are not supported for models materialized as <Term id="view">views</Term> on Spark/Databricks ([issue](https://github.com/dbt-labs/dbt-spark/issues/372))<VersionBlock lastVersion="0.21">, or Snowflake on versions older than v1.0</VersionBlock>.

Author: This is the least ugly I could manage:

<img width="870" alt="Screenshot 2022-07-05 at 14 57 48" src="https://user-images.githubusercontent.com/13897643/177333075-8045833c-c7c8-45f1-a058-19ef620998e9.png">

<img width="858" alt="Screenshot 2022-07-05 at 14 57 41" src="https://user-images.githubusercontent.com/13897643/177333062-e87ab0f2-beee-4312-b8a8-99efce064b76.png">
